app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/RecyclerViewAdapter.java

```diff
@@ -38,10 +38,10 @@
 /**
  * Implementation of {@link RecyclerView.Adapter} for {@link DisplayableItem}.
  */
-public final class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+public final class RecyclerViewAdapter<T> extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
```

Comment:
`RecyclerViewAdapter<T>` should be generic. The difference should be that it should store union type and not only `Sound` item.
This PR is first step to make it happen (making it now a bit type safe)